Autonomous_Parallel_Parking/Autonomous_Parallel_Parking.c

```diff
@@ -15,14 +15,84 @@
  ****************************************************************************************************************/
 
 #include "cpu.h"
-#include "board_led.h"
+#include "pin_setup.h"
 
-#include "foo.h"
+int count = 0;
 
 int main()
 {
   cpu_init();
-  board_led_init();
-
+  GPIOInterruptSetup();
   return 0;
 }
+
+/**
+* @brief IRQ for PA0 (user button) - to test if PA0 interrupt is functioning.  Turns on blue LED every 5 times the 
+* button is pressed.
+*/
+void EXTI0_IRQHandler(void){
```

Comment:
Yes. You know like a c file for the individual functions like a pwm.c, uart.c etc etc and then just have a state machine inside the main which calls those functions you know, like a switch inside a while(1) loop. And use the switch variable to control the state? What do you think?